CMakeLists.txt

```diff
@@ -121,6 +121,18 @@ add_library(testmodule SHARED
   "${CMAKE_CURRENT_SOURCE_DIR}/tests/testModule.cpp")
 target_link_libraries(testmodule ${Boost_LIBRARIES} ${PYTHON_LIBRARIES} Boost::${boost_python})
 
+# Get numpy dir from python, if not set externally:
+set(${numpy_include_dirs} "" CACHE PATH "Path to numpy headers")
+
+if(NOT numpy_include_dirs)
+execute_process(COMMAND "${python_interpreter}" -c "import numpy; print(numpy.get_include())"
```

Comment:
Something which is not so nice, now it finds my user-local headers:
-- Numpy header files in: /home/drothe/.local/lib/python3.8/site-packages/numpy/core/include
although boost-numpy system pacakge is found and also provides system-installed headers, under /usr/lib/python3/dist-packages/numpy/core/include
Not sure if it's easily possible to prevent this..

Reply:
Yeah but this is expected and how the python search path works, and usually what you want, you didn't install a local numpy for no reason

Reply:
what bothers me is the mixture of system lib /lib/x86_64-linux-gnu/libboost_numpy38.so.1.71.0 and numpy headers from ~/.local/lib 
It's not such a big issue since it appears only when users compile pythonbindings themselves while using both pip and installed packages. The proper solution would be that boost::numpy says where it's includes are, but it doesn't do that properly.

Reply:
I think it's fine since users can fix the issue in this case, using CACHE variable.
But please fix CACHE variable
Should be 
set(numpy_include_dirs "" CACHE PATH "Path to numpy headers")